### ISSUE TITLE
always "no cache"

### ISSUE THREAD

**Issue by zrajna** [Smeagolworms4/docker-osm-proxy#3]
Caching does not seem to work. When I deliberately load the same tile multiple times I see this in the logs:
```
[GET]  /12/2205/1225.png,
No cache call http://a.tile.openstreetmap.org/12/2205/1225.png,
[GET]  /12/2205/1225.png,
No cache call http://a.tile.openstreetmap.org/12/2205/1225.png,
[GET]  /12/2205/1225.png,
No cache call http://c.tile.openstreetmap.org/12/2205/1225.png,
[GET]  /12/2205/1225.png,
No cache call http://a.tile.openstreetmap.org/12/2205/1225.png,
```

I'm using a persistent cache (i.e. mounting /var/cache/openstreetmap-proxy folder), and I had originally trouble with writing permissions for the docker user. Now the folders and files are created, but the cache seems to be "ignored".

Any idea what is going on?

**Comment by Smeagolworms4** [comment]
Yes, the right of tour file must be similaire to uid of container run. Try
to chmod 777 on your directory or chown 1000:1000

Damien

Le ven. 12 févr. 2021 à 12:59, Zalán Rajna <notifications@github.com> a
écrit :

**Comment by zrajna** [comment]
Yes, I solved that problem (the mounted directory belongs to user 1000:1000).
However, it still does not use the cache.
(I just wrote about the permission issue because it might be related to the caching problem somehow.)

**Comment by Smeagolworms4** [comment]
The file must be written before be return by other request.
Damien


Le ven. 12 févr. 2021 à 13:20, Zalán Rajna <notifications@github.com> a
écrit :

**Comment by zrajna** [comment]
The file is correctly written on the first request, but the following 3 requests do not see the cache, even though the file is there. Or do I need to restart the container to use the cache? (That would be weird.)

**Comment by zrajna** [comment]
After a while it starts to cache properly. Is there some time that needs to elapse before caching works?
